homeassistant/components/honeywell/__init__.py

```diff
@@ -1 +1,117 @@
+from datetime import timedelta
+
+import somecomfort
+
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.util import Throttle
+
+from .const import _LOGGER, CONF_DEV_ID, CONF_LOC_ID, DOMAIN
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=180)
+PLATFORMS = ["climate"]
+
+
+async def async_setup_entry(hass, config):
+    """Set up the Honeywell thermostat."""
+    username = config.data[CONF_USERNAME]
+    password = config.data[CONF_PASSWORD]
+
+    client = await hass.async_add_executor_job(
+        get_somecomfort_client, username, password
+    )
+
+    if client is None:
+        return False
+
+    loc_id = config.data.get(CONF_LOC_ID)
+    dev_id = config.data.get(CONF_DEV_ID)
+
+    for location in client.locations_by_id.values():
+        for device in location.devices_by_id.values():
+            if (not loc_id or location.locationid == loc_id) and (
+                not dev_id or device.deviceid == dev_id
+            ):
+                data = HoneywellService(hass, client, username, password, device)
+                await data.update()
+                hass.data[DOMAIN] = data
+                hass.config_entries.async_setup_platforms(config, PLATFORMS)
+
+    return True
+
+
+def get_somecomfort_client(username, password):
+    """Initialize the somecomfort client."""
+    try:
+        return somecomfort.SomeComfort(username, password)
+    except somecomfort.AuthError:
+        _LOGGER.error("Failed to login to honeywell account %s", username)
+        return None
+    except somecomfort.SomeComfortError:
+        _LOGGER.error(
+            "Failed to initialize the Honeywell client: "
+            "Check your configuration (username, password), "
+            "or maybe you have exceeded the API rate limit?"
+        )
+        return None
+
+
+class HoneywellService:
+    """Get the latest data and update."""
+
+    def __init__(self, hass, client, username, password, device):
+        """Initialize the data object."""
+        self._hass = hass
+        self._client = client
+        self._username = username
+        self._password = password
+        self._device = device
+
+    def _retry(self) -> bool:
+        """Recreate a new somecomfort client.
+
+        When we got an error, the best way to be sure that the next query
+        will succeed, is to recreate a new somecomfort client.
+        """
+        self._client = get_somecomfort_client(self._username, self._password)
+
+        if self._client is None:
+            return False
+
+        devices = [
+            device
+            for location in self._client.locations_by_id.values()
+            for device in location.devices_by_id.values()
+            if device.name == self._device.name
+        ]
+
+        if len(devices) != 1:
+            _LOGGER.error("Failed to find device %s", self._device.name)
+            return False
+
+        self._device = devices[0]
+        return True
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
```

Comment:
In a future PR this should use `DataUpdateCoordinator`

https://developers.home-assistant.io/docs/integration_fetching_data/#coordinated-single-api-poll-for-data-for-all-entities

Reply:
For sure, I will look into that for a future PR.